inject/src/main/java/io/micronaut/inject/ast/PrimitiveElement.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.inject.ast;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.micronaut.core.annotation.AnnotationMetadata;
+
+public final class PrimitiveElement implements ClassElement {
+
+    public static final PrimitiveElement VOID = new PrimitiveElement("void", "V");
+
+    private static final PrimitiveElement INT = new PrimitiveElement("int", "I");
+    private static final PrimitiveElement CHAR = new PrimitiveElement("char", "C");
+    private static final PrimitiveElement BOOLEAN = new PrimitiveElement("boolean", "Z");
+    private static final PrimitiveElement LONG = new PrimitiveElement("long", "J");
+    private static final PrimitiveElement FLOAT = new PrimitiveElement("float", "F");
+    private static final PrimitiveElement DOUBLE = new PrimitiveElement("double", "D");
+    private static final PrimitiveElement SHORT = new PrimitiveElement("short", "S");
+    private static final PrimitiveElement BYTE = new PrimitiveElement("byte", "B");
```

Comment:
Why declaring this here if they already exist in https://github.com/micronaut-projects/micronaut-core/blob/d116782bd252f2b544594b2b55740bf86c5eb8ed/inject/src/main/java/io/micronaut/inject/writer/AbstractClassFileWriter.java#L91-L100?

Reply:
Great catch - I didn't see that map there